src/scripts/clipperUI/components/modeButtonSelector.tsx

```diff
@@ -35,12 +29,16 @@ class ModeButtonSelectorClass extends ComponentBase<{}, ClipperStateProp> {
 		if (this.props.clipperState.pageInfo.contentType !== OneNoteApi.ContentType.EnhancedUrl) {
 			return undefined;
 		}
-
-		return <ModeButton imgSrc={ExtensionUtils.getImageResourceUrl("pdf.png") }
-			label={Localization.getLocalizedString("WebClipper.ClipType.Pdf.Button")}
-			myMode={ClipMode.Pdf} tabIndex={39} selected={currentMode === ClipMode.Pdf}
-			onModeSelected={this.onModeSelected.bind(this) }
-			tooltipText={Localization.getLocalizedString("WebClipper.ClipType.Pdf.Button.Tooltip")}/>;
+		let buttonSelected: boolean = currentMode === ClipMode.Pdf;
+
+		return {
+			imgSrc: ExtensionUtils.getImageResourceUrl("pdf.png"),
+			label: Localization.getLocalizedString("WebClipper.ClipType.Pdf.Button"),
+			myMode: ClipMode.Pdf,
+			selected: buttonSelected,
+			onModeSelected: this.onModeSelected.bind(this),
+			tooltipText: Localization.getLocalizedString("WebClipper.ClipType.Pdf.Button.Tooltip")
+		};
 	}
 
 	private getAugmentationModeButton(currentMode: ClipMode) {
```

Comment:
It's not being done yet, but we should add return types to each of these methods to add type safety to the mode button props being returned

Reply:
One option is to create a generic props interface (e.g., called AriaSetItemProps) that represents an item in an aria set, and have it have these two aria properties. Props that belong to directional-key navigatable components can then have their props extend that interface.

Reply:
The new ModeButtonProps type can then be a union of those two types:
e.g., ModeButtonProps = AriaSetItemProps & ModeButtonPropsWithoutAriaSetItemProps (or whatever name is better). This method can then return ModeButtonPropsWithoutAriaSetItemProps